diff -Naur packages.orig/nose//result.py site-packages/nose/result.py
--- packages.orig/nose//result.py	2014-03-05 12:23:14.000000000 +0800
+++ site-packages/nose/result.py	2014-03-11 10:51:18.000000000 +0800
@@ -43,14 +43,6 @@
         self.config = config
         _TextTestResult.__init__(self, stream, descriptions, verbosity)
 
-    def addSkip(self, test, reason):
-        # 2.7 skip compat
-        from nose.plugins.skip import SkipTest
-        if SkipTest in self.errorClasses:
-            storage, label, isfail = self.errorClasses[SkipTest]
-            storage.append((test, reason))
-            self.printLabel(label, (SkipTest, reason, None))
-
     def addError(self, test, err):
         """Overrides normal addError to add support for
         errorClasses. If the exception is a registered class, the
